AlfrescoApp/Utils/KeychainUtils.m

```diff
@@ -102,6 +103,37 @@ + (BOOL)updateSavedAccounts:(NSArray *)accounts forListIdentifier:(NSString *)li
     return updateSucceeded;
 }
 
++ (BOOL)updateSavedAccount:(UserAccount *)account forListIdentifier:(NSString *)listIdentifier error:(NSError *__autoreleasing *)updateError
+{
+    BOOL updateSucceeded = YES;
+    
+    if(account)
+    {
+        NSMutableArray *accountsList = [[self savedAccountsForListIdentifier:listIdentifier error:updateError] mutableCopy];
+        if(accountsList)
+        {
+            // Get the account
+            NSPredicate *predicate = [NSPredicate predicateWithFormat:@"accountIdentifier == %@", account.accountIdentifier];
+            NSArray *accountArray = [accountsList filteredArrayUsingPredicate:predicate];
+            UserAccount *keychainAccount = accountArray.firstObject;
+            NSUInteger index = [accountsList indexOfObject:keychainAccount];
+            [accountsList replaceObjectAtIndex:index withObject:account];
```

Comment:
In the very unlikely scenario of accountArray being empty, then the index would not be within the bounds of the accountsList array, thus triggering a crash.